### PR TITLE
Pv48 share image detail list user

### PR DESCRIPTION
**Pull request by Feyzarkt**

### CODE REVIEW

**Review by ibrahimertanylmz — COMMENTED** [review]
Great work!

File: app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt [hunk 1]
```diff
@@ -109,7 +108,6 @@ class CloudDbRepository @Inject constructor(
         cloudDB.createObjectType(ObjectTypeInfoHelper.getObjectTypeInfo())
     }
 
-    @ExperimentalCoroutinesApi
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt:112]: Does this change effect project in some conditions? If we are sure about it than it's fine to remove this

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/shareimagedetail/ShareImageDetailViewModel.kt [hunk 2]
```diff
@@ -166,7 +165,6 @@ class ShareImageDetailViewModel @Inject constructor(
         }
     }
 
-    @OptIn(ExperimentalCoroutinesApi::class)
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/shareimagedetail/ShareImageDetailViewModel.kt:175]: Same condition check for this line

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserFragment.kt [hunk 3]
```diff
@@ -87,6 +75,18 @@ class ListUserFragment : DialogFragment() {
         listUserAdapter.onUserSelectListener {
             viewModel.selectUser(it)
         }
+        binding.saveValuesButton.setOnClickListener {
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserFragment.kt:78]: you can override setupListeners function and move listeners inside it if it's possible

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserFragment.kt [hunk 4]
```diff
@@ -87,6 +75,18 @@ class ListUserFragment : DialogFragment() {
         listUserAdapter.onUserSelectListener {
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserFragment.kt:75]: you can override setupListeners function and move listeners inside it if it's possible

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserViewModel.kt [hunk 5]
```diff
@@ -98,44 +97,49 @@ class ListUserViewModel @Inject constructor(
...
+        }
     }
 
     fun errorShown() {
         _listUserUiState.update {
             it.copy(error = null)
         }
     }
+
+    companion object{
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserViewModel.kt]: error messages should extract to resources for future usage, translation etc. So you can get from resources using provider inside viewmodel

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/shareimagedetail/ShareImageDetailFragment.kt [hunk 6]
```diff
@@ -45,20 +44,40 @@ class ShareImageDetailFragment :
     @Inject
     lateinit var sharedUsersAdapter: SharedUsersAdapter
 
+    private lateinit var alertDialogSharedPeople: AlertDialog
+
+    companion object{
+        const val PATH_IMAGE = "image/*"
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/shareimagedetail/ShareImageDetailFragment.kt]: path should extract to Constant.kt

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/shareimagedetail/ShareImageDetailViewModel.kt [hunk 7]
```diff
@@ -256,13 +248,15 @@ class ShareImageDetailViewModel @Inject constructor(
...
-            Log.d("compressed-last", getImageSize(baos).toString())
             baos.toByteArray()
         }
 
 
     private fun getImageSize(baos: ByteArrayOutputStream) = baos.toByteArray().size / 1024
+
+    companion object{
+        const val ERROR_SELECT_PHOTO = "Please select photo for sharing.."
+        const val ERROR_DID_NOT_UPLOAD = " image did not upload. Please try again."
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/shareimagedetail/ShareImageDetailViewModel.kt]: should extract to resources and can get from resources using provider

File: app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserFragment.kt [hunk 8]
```diff
@@ -42,11 +42,15 @@ class ListUserFragment : DialogFragment() {
     @Inject
     lateinit var listUserAdapter: ListUserAdapter
 
+    companion object{
+        const val REQUEST_KEY = "requestKey"
+        const val BUNDLE_KEY = "bundleKey"
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/ui/listuser/ListUserFragment.kt]: keys should be placed in Constant.kt

File: app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt [hunk 9]
```diff
@@ -569,5 +559,8 @@ class CloudDbRepository @Inject constructor(
         private const val DB_NAME = "PhotoAppDB"
         private const val TAG = "CloudDB"
         private const val FILE_ID = "fileId"
+        private const val ID = "id"
+        private const val EXP_STH_WENT_WRONG = "Something went wrong"
+        private const val EXP_STH_WENT_WRONG_DELETING = "Something went wrong during deleting"
```
Comment (ibrahimertanylmz) [inline app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt]: strings should be extracted to resources

File: app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt (hunk 1 above)
Comment (Feyzarkt) [reply app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt:112]: When I tested the relevant parts, I didn't have a problem and I saw such an expression in Stackoverflow, I don't think we need to use it anymore. "UPDATE: Kotlin coroutines are no longer experimental as of Kotlin 1.3."

Comment (ibrahimertanylmz) [reply app/src/main/java/com/hms/referenceapp/photoapp/data/repository/CloudDbRepository.kt:112]: I got it, thanks!!

**Pull request merged**